packages/dd-trace/src/appsec/telemetry/telemetry-collector.js

```diff
@@ -0,0 +1,117 @@
+'use strict'
+
+const { aggregated, conflated, delegating } = require('./handlers')
+const log = require('../../log')
+const DD_TELEMETRY_COLLECTOR = Symbol('_dd.appsec.telemetryCollector')
+
+class TelemetryCollector {
+  constructor (builders) {
+    this.handlers = new Map()
+    this.builders = builders
+  }
+
+  addMetric (metric, value, tag) {
+    this.getOrCreateHandler(metric).add(value, tag)
+  }
+
+  getOrCreateHandler (metric) {
+    let handler = this.handlers.get(metric)
+    if (!handler) {
+      handler = this.builders(metric)
+      this.handlers.set(metric, handler)
+    }
+    return handler
+  }
+
+  drainMetrics () {
+    const result = []
+    for (const handler of this.handlers.values()) {
+      const values = handler.drain()
+      if (values && values.length) {
+        result.push(...values)
+      }
+    }
+    this.handlers.clear()
+    return result
+  }
+
+  merge (metrics) {
+    if (metrics) {
+      for (const metricData of metrics) {
+        this.getOrCreateHandler(metricData.metric).merge(metricData)
+      }
+    }
+  }
+
+  reset () {
+    this.handlers = new Map()
+  }
+}
+
+const GLOBAL = new TelemetryCollector(metric => metric.hasRequestScope()
```

Comment:
It doesn't look very descriptive... and it can be confusing with `global`, maybe `globalTelemetryCollector`?